### PR TITLE
Check the result of PySet_Contains() for error in Python/symtable.c

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

File: Python/symtable.c [hunk 1]
```diff
@@ -543,14 +544,22 @@ analyze_name(PySTEntryObject *ste, PyObject *scopes, PyObject *name, long flags,
                          "nonlocal declaration not allowed at module level");
             return error_at_directive(ste, name);
```
Comment (vstinner) [inline Python/symtable.c:545]: Remark unrelated to you PR. I don't understand why an exception is raised before calling error_at_directive() which seems to always override the current exception (and not chain the new one to the currently raised one).

**Review by vstinner — APPROVED** [review]
LGTM.

File: Python/symtable.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Python/symtable.c:545]: No, it only sets location in the raised SyntaxError.

Comment (vstinner) [reply Python/symtable.c:545]: I looked again at PyErr_SyntaxLocationObjectEx(): oh, you're right. Now it makes sense, thanks :-)

**Comment by vstinner** [comment]
Nice fix, it's good to check for errors :-)

**Pull request merged**